addon/mixins/route.js

```diff
@@ -283,17 +278,61 @@ export default Ember.Mixin.create({
 
    @method updateInfinityModel
    @param {Ember.Enumerable} newObjects The new objects to add to the model
-   @return {Ember.Array} returns the updated infinity model
+   @return {DS.RecordArray} returns the updated infinity model
    */
   updateInfinityModel(newObjects) {
```

Comment:
I agree that we shouldn't introduce breaking changes in this PR, but I would almost advocate for deprecating the current usage of `updateInfinityModel`since it's inconsistent, and moving towards a new hook.

I'd also advocate for moving `currentPage` into the new `updateBoundParams` (or whatever) hook, and treating it as just another bound param. In our cursor-based infinity list, for example, the server doesn't even use the page number, it's just along for the ride.